src/app/features/public-form/public-form-permission.service.spec.ts

```diff
@@ -0,0 +1,156 @@
+      public: [{ subject: "Child", action: "create" }],
+    });
+    mockEntityMapper.load.and.resolveTo(permissionsConfig);
+
+    const result = await service.hasPublicCreatePermission("Child");
+
+    expect(result).toBeTrue();
+  });
+
+  it("should allow access when public role has manage permission", async () => {
+    const permissionsConfig = new Config(Config.PERMISSION_KEY, {
+      public: [{ subject: "Child", action: "manage" }],
+    });
+    mockEntityMapper.load.and.resolveTo(permissionsConfig);
+
+    const result = await service.hasPublicCreatePermission("Child");
+
+    expect(result).toBeTrue();
+  });
+
+  it("should deny access when public role has no permission for the entity type", async () => {
+    const permissionsConfig = new Config(Config.PERMISSION_KEY, {
+      public: [{ subject: "School", action: "create" }],
+    });
+    mockEntityMapper.load.and.resolveTo(permissionsConfig);
+
+    const result = await service.hasPublicCreatePermission("Child");
+
+    expect(result).toBeFalse();
+  });
+
+  it("should detect admin permission when user has admin_app role", () => {
+    mockSessionSubject.next({ roles: ["admin_app", "user"] });
+
+    const result = service.hasAdminPermission();
+
+    expect(result).toBeTrue();
+  });
+
+  it("should not detect admin permission when user lacks admin_app role", () => {
+    mockSessionSubject.next({ roles: ["user", "viewer"] });
+
+    const result = service.hasAdminPermission();
+
+    expect(result).toBeFalse();
+  });
+
+  it("should not detect admin permission when no session exists", () => {
+    mockSessionSubject.next(null);
+
+    const result = service.hasAdminPermission();
+
+    expect(result).toBeFalse();
+  });
+
+  it("should create new permissions config when none exists", async () => {
+    mockEntityMapper.load.and.rejectWith(new Error("Config not found"));
+    mockEntityMapper.save.and.resolveTo(undefined);
+
+    await service.addPublicCreatePermission("Child");
+
+    expect(mockEntityMapper.save).toHaveBeenCalledWith(
+      jasmine.objectContaining({
+        data: {
+          public: [{ subject: "Child", action: "create" }],
+        },
+      }),
+      true,
+    );
+  });
+
+  it("should add permission to existing permissions config", async () => {
+    const existingConfig = new Config(Config.PERMISSION_KEY, {
+      public: [{ subject: "School", action: "create" }],
+    });
+    mockEntityMapper.load.and.resolveTo(existingConfig);
+    mockEntityMapper.save.and.resolveTo(undefined);
+
+    await service.addPublicCreatePermission("Child");
+
+    expect(mockEntityMapper.save).toHaveBeenCalledWith(
+      jasmine.objectContaining({
+        data: {
+          public: [
+            { subject: "School", action: "create" },
+            { subject: "Child", action: "create" },
+          ],
+        },
+      }),
+      true,
+    );
+  });
+
+  it("should skip adding permission when it already exists", async () => {
```

Comment:
Do we have a test case for public "read" (but not create) permission? That also has to trigger the warning and add permission

Reply:
works :+1: